### PR TITLE
Make HTTP::Server less stupid about unexpected errors

### PR DESCRIPTION
**Pull request by jhass**

### CODE REVIEW

**Comment by bcardiff** [comment]
In which case the LogHandler should swallow the exception?

**Comment by jhass** [comment]
In the case that you just want it logged and not handle it anymore.

**Comment by bcardiff** [comment]
But I mean, why that should be tided with the logger? In the default behavior you don't swallow, which is also the current behavior. If stoping the exception in the call chain is desired I would expect a different handler than the logger.

**Comment by jhass** [comment]
Alright, I made the `ErrorHandler` not spill the guts to the client by default instead.

**Comment by bcardiff** [comment]
Thanks @jhass , `ErrorHandler` is the right place. 👍 from my side here.

File: spec/std/http/server/handlers/error_handler_spec.cr [hunk 1]
```diff
@@ -8,7 +8,7 @@ describe HTTP::ErrorHandler do
     response = HTTP::Server::Response.new(io)
     context = HTTP::Server::Context.new(request, response)
 
-    handler = HTTP::ErrorHandler.new
+    handler = HTTP::ErrorHandler.new(true)
```
Comment (ysbaddaden) [inline spec/std/http/server/handlers/error_handler_spec.cr]: What about `verbose: true` ? I had to search to understand what `true` meant.

Comment (jhass) [inline spec/std/http/server/handlers/error_handler_spec.cr]: Done.

**Comment by jhass** [comment]
Anything else needed here?

File: src/http/server/response.cr [hunk 2]
```diff
@@ -104,6 +104,17 @@ class HTTP::Server
       @output.close
     end
 
+    # Generates an error response using *message* and *code*.
+    #
+    # Calls `reset` and then writes the given message.
+    def respond_with_error(message = "Internal Server Error", code = 500)
+      reset
+      @status_code = code
+      @content_type = "text/plain"
```
Comment (asterite) [inline src/http/server/response.cr]: I think this should be `self.content_type = "text/plain"`, there's no `@content_type` variable (it's in the headers)

Comment (asterite) [inline src/http/server/response.cr]: Other than this, it looks good for me

Comment (jhass) [inline src/http/server/response.cr]: Duh, good catch, fixed.

**Pull request merged**